adapter-mongodb/src/main/java/io/nosqlbench/adapter/mongodb/core/MongodbDriverAdapter.java

```diff
@@ -18,33 +18,64 @@
 
 import io.nosqlbench.api.config.standard.NBConfigModel;
 import io.nosqlbench.api.config.standard.NBConfiguration;
+import io.nosqlbench.api.config.standard.Param;
 import io.nosqlbench.engine.api.activityimpl.OpMapper;
 import io.nosqlbench.engine.api.activityimpl.uniform.BaseDriverAdapter;
 import io.nosqlbench.engine.api.activityimpl.uniform.DriverAdapter;
 import io.nosqlbench.engine.api.activityimpl.uniform.flowtypes.Op;
 import io.nosqlbench.nb.annotations.Service;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
 import java.util.function.Function;
 
 /**
  * Special thanks to Justin Chu who authored the original NoSQLBench MongoDB ActivityType.
  */
-@Service(value=DriverAdapter.class, selector ="mongodb")
+@Service(value = DriverAdapter.class, selector = "mongodb")
 public class MongodbDriverAdapter extends BaseDriverAdapter<Op, MongoSpace> {
 
+    private static final Logger logger = LogManager.getLogger(MongodbDriverAdapter.class);
+    private String connection;
+
     @Override
     public OpMapper<Op> getOpMapper() {
         return new MongoOpMapper(this);
     }
 
     @Override
     public Function<String, ? extends MongoSpace> getSpaceInitializer(NBConfiguration cfg) {
-        return s -> new MongoSpace(s, cfg);
+        return s -> new MongoSpace(s, this.connection, cfg);
     }
 
     @Override
     public NBConfigModel getConfigModel() {
-        return super.getConfigModel().add(MongoSpace.getConfigModel());
+
+        NBConfigModel mongoConfigModel = MongoSpace.getConfigModel();
+        NBConfigModel parentModel = super.getConfigModel();
+        final NBConfigModel merged = mongoConfigModel.add(parentModel);
+
+        if (logger.isDebugEnabled()) {
```

Comment:
This specialized debug logic having to do with NBConfig* mechanisms should be moved into it or removed altogether. It would be even better as new unit tests and javadoc.